pkg/liquidity-source/balancer-v2/math/fixed_point.go

```diff
@@ -128,6 +128,67 @@ func (l *fixedPoint) PowUp(x *uint256.Int, y *uint256.Int) (*uint256.Int, error)
 	return l.Add(raw, maxError)
 }
 
+func (l *fixedPoint) PowDown(x *uint256.Int, y *uint256.Int) (*uint256.Int, error) {
+	if y.Eq(l.ONE) {
+		return x, nil
+	}
+	if y.Eq(l.TWO) {
+		return l.MulUp(x, x)
+	}
+	if y.Eq(l.FOUR) {
+		square, err := l.MulUp(x, x)
+		if err != nil {
+			return nil, err
+		}
+
+		return l.MulUp(square, square)
+	}
+
+	raw, err := LogExpMath.Pow(x, y)
+	if err != nil {
+		return nil, err
+	}
+
+	mulUpRawAndMaxPow, err := l.MulUp(raw, l.MAX_POW_RELATIVE_ERROR)
+	if err != nil {
+		return nil, err
+	}
+
+	maxError, err := l.Add(mulUpRawAndMaxPow, number.Number_1)
+	if err != nil {
+		return nil, err
+	}
+
+	if raw.Lt(maxError) {
+		return number.Zero, nil
+	}
+
+	return l.Sub(raw, maxError)
```

Comment:
```suggestion
	return l.PowDownV1(x, y)
```

Why do we have both `PowDown` and `PowDownV1`? 
I think we should either keep only one version if there's no significant difference, or reuse `PowDownV1` here to avoid code duplication.